private/findcluster.m

```diff
@@ -82,26 +83,33 @@
 % for each channel (combination), find the connected time-frequency clusters
 labelmat = zeros(size(onoff));
 total = 0;
-if nfreq*ntime>1
+if prod(ndatadim)>1
   for spatdimlev=1:spatdimlength
-    [labelmat(spatdimlev, :, :), num] = spm_bwlabel(double(reshape(onoff(spatdimlev, :, :), nfreq, ntime)), 6); % the previous code contained a '4' for input
-    labelmat(spatdimlev, :, :) = labelmat(spatdimlev, :, :) + (labelmat(spatdimlev, :, :)~=0)*total;
+    if numel(ndatadim) == 2 || numel(ndatadim) == 3 % if 2D or 3D data (without channel)
+      % use spm_bwlabel for 2D/3D data to avoid usage of image processing toolbox
+      [labelmat(spatdimlev, :, :,:), num] = spm_bwlabel(double(reshape(onoff(spatdimlev, :, :), celldims{:})), 6); % the previous code contained a '4' for input
+    else
+      ft_error('cluster-based stats for data >4D is NYI')
+    end
+      
+    
+    labelmat(spatdimlev, :, :, :) = labelmat(spatdimlev, :, :, :) + (labelmat(spatdimlev, :, :, :)~=0)*total;
     total = total + num;
   end
 else
   labelmat(onoff>0) = 1:sum(onoff(:));
   total = sum(onoff(:));
 end
 % combine the time and frequency dimension for simplicity
-labelmat = reshape(labelmat, spatdimlength, nfreq*ntime);
+labelmat = reshape(labelmat, spatdimlength, prod(ndatadim));
 
 % combine clusters that are connected in neighbouring channel(s)
 % (combinations). Convert inputs to uint32 as that is required by the mex
 % file (and the values will be positive integers anyway).
 cluster = combineClusters(uint32(labelmat), logical(spatdimneighbstructmat), uint32(total));
 
 % reshape the output to the original format of the data
-cluster = reshape(cluster, spatdimlength, nfreq, ntime);
+cluster = reshape(cluster, spatdimlength, celldims{:});
```

Comment:
OK, the changes to this function look clean enough. We look forward to some code added to an existing test function, or a new test function which highlights your use case.